### PR TITLE
Remove Pytorch Dependency

### PR DESCRIPTION
**Pull request by Grahams-Uncle**
I have tested it using how_to_use_cfnet.ipynb, it works the same as before. Let me know if it has any issue.

### CODE REVIEW

**Comment by Grahams-Uncle** [comment]
There was some issue with the checks for previous commits, but it was fixed for the latest one.

File: cfnet/datasets.py [hunk 1]
```diff
@@ -32,23 +31,52 @@ def _numpy_collate(batch):
   else:
     return np.array(batch)
 
-class NumpyLoader(DataLoader):
-  def __init__(self, dataset, batch_size=1,
+class DataLoader:
+    def __init__(self,dataset, batch_size=1,
```
Comment (BirkhoffG) [inline cfnet/datasets.py:35]: Please document the meaning of each argument

See how to write Jupyter Notebook documents:
https://birkhoffg.github.io/cfnet/tutorials/contribution.html#write-documentations-in-jupyter-notebook

File: cfnet/datasets.py [hunk 2]
```diff
@@ -32,23 +31,52 @@ def _numpy_collate(batch):
...
+        self.sampler=sampler
+        self.batch_sampler=batch_sampler
+        self.num_workers=num_workers
+        self.collate_fn=collate_fn
+        self.pin_memory=pin_memory
+        self.drop_last=drop_last
+        self.timeout=timeout
+        self.worker_init_fn=worker_init_fn
+
+        self.dataLen = len(dataset)
```
Comment (BirkhoffG) [inline cfnet/datasets.py:53]: Use the undercase naming, instead of the camelcase, e.g., name variables as `data_len`. 

Python defines variables in undercase

File: cfnet/datasets.py [hunk 3]
```diff
@@ -32,23 +31,52 @@ def _numpy_collate(batch):
...
+        self.keySeq = hk.PRNGSequence(42)
+        self.keySeq.reserve(len(self))
+        self.key = next(self.keySeq)
+        self.indices = jax.numpy.arange(self.dataLen)
+        self.pose = 0
+    def __len__(self):
+        batches = -(len(self.dataset) // -self.batch_size)
```
Comment (BirkhoffG) [inline cfnet/datasets.py:60]: Why are you using `-`. If not necessary, please remove them.

File: cfnet/datasets.py [hunk 4]
```diff
@@ -32,23 +31,52 @@ def _numpy_collate(batch):
...
+        self.batch_sampler=batch_sampler
+        self.num_workers=num_workers
+        self.collate_fn=collate_fn
+        self.pin_memory=pin_memory
+        self.drop_last=drop_last
+        self.timeout=timeout
+        self.worker_init_fn=worker_init_fn
+
+        self.dataLen = len(dataset)
+        self.keySeq = hk.PRNGSequence(42)
```
Comment (BirkhoffG) [inline cfnet/datasets.py:54]: Pass the seed (i.e., 42 here) as an argument in the `__init__`. It should have a default value. (e.g., 42).

**Review by BirkhoffG — CHANGES_REQUESTED** [review]
Please fix these comments.

Also, please provide some **test cases** for `Dataloader`. These test cases can be the code for you to test the correctness of `Dataloader`.

See https://birkhoffg.github.io/cfnet/tutorials/contribution.html#write-test-cases-in-jupyter-notebook

File: cfnet/datasets.py (hunk 3 above)
Comment (Grahams-Uncle) [reply cfnet/datasets.py:60]: Use - to get the ceil instead of floor.

Comment (BirkhoffG) [reply cfnet/datasets.py:61]: Give a brief comment of this line

**Comment by Grahams-Uncle** [comment]
fix #37

File: cfnet/datasets.py [hunk 5]
```diff
@@ -32,23 +31,63 @@ def _numpy_collate(batch):
   else:
     return np.array(batch)
 
-class NumpyLoader(DataLoader):
-  def __init__(self, dataset, batch_size=1,
-                shuffle=False, sampler=None,
-                batch_sampler=None, num_workers=0,
+class DataLoader:
+    def __init__(self,dataset, batch_size=1,
```
Comment (BirkhoffG) [inline cfnet/datasets.py:35]: Here, you should document each argument by writing a comment right after them.

For example:
```python
def __init__(
    self,
    dataset: Dataset, # dataset, a Dataset object
    batch_size: int=1, # batch size
    ...
)
```

Copy-paste all your comments right after each argument.

When running `nbdev_preview`, you will see the documents.

**Review by BirkhoffG — CHANGES_REQUESTED** [review]
Everything else looks good, except you should move comments after each argument.

You can view the documentation via `nbdev_preview`.

**Review by BirkhoffG — APPROVED** [review]
Looks good to me
